app/forms/users/edit_user_form.rb

```diff
@@ -23,4 +26,23 @@ def user_role_options(roles = User.roles.keys)
       OpenStruct.new(label: I18n.t("users.roles.#{role}.name"), value: role, description: I18n.t("users.roles.#{role}.description"))
     end
   end
+
+  def user_organisation_slug_options(organisations = nil)
+    organisations ||= available_user_organisations
+    organisations.map do |slug, name|
+      OpenStruct.new(value: slug, label: name)
+    end
+  end
+
+private
+
+  def available_user_organisations
+    organisations = Organisation.all.pluck(:slug, :name)
+    organisations.sort_by! { |slug, _| slug }
+    organisations
```

Comment:
> Couple of minor syntax questions here:
> 
> 1. Do you need the `_` argument for `sort_by!` or can you just do `{ |slug| slug }`

So what I thought I was doing with `|slug, _|` was using Array decomposition to get the first element of each Array in the Array of organisations (because organisations would be something like `[['test-org', 'Test Org'], ['department-for-tests', 'Department For Tests']]`). That is what is happening, but it turns out that a) the documentation for Ruby suggests you need to put parentheses around the arguments for it to work (so `|(slug, _)|`) [[1](https://docs.ruby-lang.org/en/3.0/syntax/methods_rdoc.html#label-Array+Decomposition)] b) the definition of the `<=>` operator for Array means that a normal `sort!` would probably work just as well, as it tests each element in each Array in turn [[2](https://docs.ruby-lang.org/en/3.0/Array.html#method-i-3C-3D-3E)] (and we're not going to have two orgs with the same slug).

I'm kind of torn, I prefer the explicitness of saying "we're going to sort by slug", but if it isn't obvious from what I've written already then maybe it doesn't help much.

> 2. Is it possible to return `organisations.sort_by! { |slug, _| slug }`, or do you need the `organisations` line?

Yes, you're right, `organisations` line is not necessary, `sort_by!` returns `self`.

Reply:
I think what you've said about being explicit about sorting by slug makes sense. I think I'd have understood it / been more likely to look up the decomposition syntax with the parentheses there.  But as you say, it's probably easier (and I think possibly faster?) to do the ordering on the DB side.